src/library_mvc/src/js/model.js

```diff
@@ -0,0 +1,141 @@
+function Model(userStore) {
+  this.userStore = userStore;
+  this.books = null;
+  this.state = {
+  books: null,
+  history: null,
+  search: null,
+  curentFilter: null,
+  }
+}
+
+Model.prototype.getHistory = function() {
+  let historyList = [];
+  let sortedArray = this.state.history.sort(function (a, b) {
+    return b.time - a.time
+  });
+  if (sortedArray.length < 3) {
+    for (let i = 0; i < sortedArray.length; i++) {
+      historyList.push(sortedArray[i]);
+    }
+  } else {
+    for (let i = 0; i < 3; i++) {
+      historyList.push(sortedArray[i]);
+    }
+  }
+  return(historyList);
+};
+
+Model.prototype.setActionHistory = function(action, details) {
+  if (action && details) {
+    let historyElement = {
+    time: Date.now(),
+    type: action,
+    details: details
+    };
+    this.state.history.push(historyElement);
+    if (this.userStore.get() === null) {
+      this.userStore.set(this.state.history);
+    } else {
+      this.state.history = this.userStore.get();
+      this.state.history.push(historyElement);
+      this.userStore.set(this.state.history);
+    }
+  } else {
+    if (this.userStore.get() !== null) {
+      this.state.history = this.userStore.get();
+    }
+  }
+};
+
+Model.prototype.createBook = function(elem) {
+  if (elem) {
+    let book = {
+      id: elem.id || getUniqueId(),
+      title: elem.title,
+      author: {
+        firstName: elem.author.firstName,
+        lastName: elem.author.lastName
+      },
+      cost: elem.cost,
+      image_url: elem.image_url,
+      rating: elem.rating,
+      categories: elem.rating,
+      createdAt: elem.createdAt || new Date().getTime(),
+      updatedAt: elem.updatedAt || new Date().getTime()
+    };
+    this.books.push(book);
+  }
+};
+
+Model.prototype.changePropertyBook = function(id, property, value) {
+  let requiredBook = this.books.findIndex((elem) => {
```

Comment:
`let requiredBook = this.books.find((elem) => elem.id === id)
requiredBook[property] = value;`